mvc/src/main/java/nextstep/mvc/controller/tobe/AnnotationHandlerMapping.java

```
@@ -21,10 +32,58 @@ public AnnotationHandlerMapping(final Object... basePackage) {
     }
 
+    private void initHandlerExecutions(final Object[] basePackage) {
+        final Set<Class<?>> controllerClasses = findControllerClasses(basePackage);
+        controllerClasses.forEach(
+                controller -> addExecutions(newInstance(controller), findRequestMappingMethods(controller))
+        );
+    }
+
+    private Set<Class<?>> findControllerClasses(final Object[] basePackage) {
+        final Reflections reflections = new Reflections(basePackage);
+        return reflections.getTypesAnnotatedWith(Controller.class);
+    }
+
+    private Object newInstance(final Class<?> clazz) {
+        try {
+            final Constructor<?> constructor = clazz.getConstructor();
+            return constructor.newInstance();
+        } catch (final Exception e) {
+            throw new NoDefaultConstructorException();
+        }
+    }
+
+    private List<Method> findRequestMappingMethods(final Class<?> clazz) {
+        final Method[] methods = clazz.getDeclaredMethods();
+        return Arrays.stream(methods)
+                .filter(it -> it.isAnnotationPresent(RequestMapping.class))
+                .collect(Collectors.toList());
+    }
+
+    private void addExecutions(final Object controllerInstance, final List<Method> requestMappingMethods) {
+        for (final Method method : requestMappingMethods) {
+            addExecution(controllerInstance, method);
+        }
+    }
+
+    private void addExecution(final Object controllerInstance, final Method method) {
+        final RequestMapping annotation = method.getAnnotation(RequestMapping.class);
+        for (final RequestMethod requestMethod : annotation.method()) {
+            final HandlerKey key = new HandlerKey(annotation.value(), requestMethod);
+            handlerExecutions.put(key, new HandlerExecution(controllerInstance, method));
+        }
+    }
+
     public Object getHandler(final HttpServletRequest request) {
-        return null;
+        Objects.requireNonNull(request);
```

request가 null일때 어떻게 되나요?

---

request가 null이면 Objects.requireNonNull에서 `NullPointerException` 을 터뜨립니다.
request는 서블릿 외부에서 주입시켜주기 때문에 null일 수가 없긴 하지만 혹시나 하는 마음에 null 처리를 추가해보았습니다.
널을 미리 처리함으로써 fail fast로 안정적인 프로그램을 짤 수 있다고 생각합니다.